lib/EarlyFunctionAnalysis/DetectABI.cpp

```diff
@@ -863,6 +868,9 @@ DetectABI::buildPrototypeForIndirectCall(const FunctionSummary &CallerSummary,
   auto NewType = makeType<RawFunctionType>();
   auto &CallType = *llvm::cast<RawFunctionType>(NewType.get());
   {
+    revng_assert(Binary->Architecture() != Architecture::Invalid);
+    CallType.Architecture() = Binary->Architecture();
```

Comment:
You can infer the architecture from the `MetaAddress` of `CallerBlock.Begin()`.

Please review all the places where we assign `RawFunctionType::Architecture` and let's make sure we resort to `Binary::Architecture` only in the extreme case. In general, there should be a more "local" way of inferring the architecture.

Reply:
There is no `CallerBlock.Begin()`, but there is `CallerBlock.End()`. I suppose that will work too?

Reply:
`.ID().start()`